internal/gengapic/gengapic_test.go

```diff
@@ -378,6 +349,59 @@ methods:
 	}
 }
 
+func TestMethodDoc(t *testing.T) {
+	m := &descriptor.MethodDescriptorProto{
+		Name: proto.String("MyMethod"),
+	}
+
+	var g generator
+	g.comments = make(map[proto.Message]string)
+
+	for _, tst := range []struct {
+		in, want   string
+		deprecated bool
+	}{
+		{
+			in:   "",
+			want: "",
+		},
+		{
+			in:   "Does stuff.\n It also does other stuffs.",
+			want: "// MyMethod does stuff.\n// It also does other stuffs.\n",
+		},
+		{
+			in:         "Does not have a proper comment.",
+			want:       "// MyMethod does not have a proper comment.\n//\n// Deprecated: This may be removed in a future version.\n",
+			deprecated: true,
+		},
+		{
+			in:         "Deprecated: this is a proper deprecation notice.",
+			want:       "// Deprecated: this is a proper deprecation notice.\n",
+			deprecated: true,
+		},
+		{
+			in:         "This is a comment that includes Deprecated but not at the beginning.",
+			want:       "// MyMethod this is a comment that includes Deprecated but not at the beginning.\n",
```

Comment:
I think we still might want a deprecated line here. We should double check to see what the common Go linters check for when reading deprecation notices.

Reply:
https://github.com/golang/go/wiki/Deprecated

I think we need the "// Deprecated:"

Reply:
So a proto comment (which in the generator doesn't have `//`) would need to have some line that begins with `"Deprecated:"` in order for us to _not_ generate our own generic message?

Reply:
I believe so.

Reply:
Thanks for catching. Done.